server.go

```diff
@@ -51,15 +51,16 @@ type Server struct {
 	// The transport passed to `http.Client` when fetching fragments or proxying
 	// requests.
 	HttpTransport http.RoundTripper
-	// A function that is called before the request is handled by viewproxy.
-	// Call the callback to have viewproxy handle the request.
-	AroundRequest func(w http.ResponseWriter, r *http.Request, callback func())
+	// A function to wrap request handling with other middleware
+	AroundRequest func(http.Handler) http.Handler
 	tracingConfig tracing.TracingConfig
 	// A function that is called when an error occurs in the viewproxy handler
 	OnError func(w http.ResponseWriter, r *http.Request, e error)
 }
 
-const RouteContextKey = "viewproxy-route"
+type routeContextKey struct{}
+
+type parametersContextKey struct{}
```

Comment:
For my own understanding, is it recommended to use `struct{}` over `string` for context keys?

I guess I would have thought that both keys would return the same value since they're empty structs.

Reply:
Yeah, I think the linter references providing a collision free key which the internal struct provides.

Reply:
I see, that makes sense. today I learned ✨